### PR TITLE
deps: update dockerfile dependencies

### PR DESCRIPTION
**Pull request by chenrui333**

### CODE REVIEW

**Comment by yxxhero** [comment]
@chenrui333 Thanks for your work. Please fix the ci issue.

**Comment by mumoshu** [comment]
@yxxhero I just realized that we missed covering the supported kustomize versions in our [test matrix](https://github.com/helmfile/helmfile/blob/main/.github/workflows/ci.yaml). We shall probably figure out which versions to support and modify the workflow accordingly, in a separate PR. WDYT?

**Comment by yxxhero** [comment]
@mumoshu yeah. I think so. I will create a PR for it.

**Comment by yxxhero** [comment]
ping

File: pkg/helmexec/exec_test.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ exec: helm --kube-context dev repo add myRepo https://repo.example.com/
 	err = helm.AddRepo("acrRepo", "", "", "", "", "", "", "acr", "", "")
 	expected = `Adding repo acrRepo (acr)
 exec: az acr helm repo add --name acrRepo
-exec: az acr helm repo add --name acrRepo: 
+exec: az acr helm repo add --name acrRepo:
```
Comment (yxxhero) [inline pkg/helmexec/exec_test.go:165]: `exec: az acr helm repo add --name acrRepo: `

Comment (yxxhero) [inline pkg/helmexec/exec_test.go:165]:
```suggestion
exec: az acr helm repo add --name acrRepo: 
```

File: Dockerfile.debian-stable-slim [hunk 2]
```diff
@@ -37,18 +37,18 @@ RUN set -x && \
...
-# the sha256 sum can be found at https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl.sha256
+# the sha256 sum can be found at https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl.sha256
 # maybe a good idea to automate in the future?
-ENV KUBECTL_VERSION="v1.21.4"
-ENV KUBECTL_SHA256="9410572396fb31e49d088f9816beaebad7420c7686697578691be1651d3bf85a"
+ENV KUBECTL_VERSION="v1.25.2"
+ENV KUBECTL_SHA256="8639f2b9c33d38910d706171ce3d25be9b19fc139d0e3d4627f38ce84f9040eb"
 RUN set -x && \
-    wget "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
+    wget "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
```
Comment (yxxhero) [inline Dockerfile.debian-stable-slim:45]:
```suggestion
    wget "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
```

File: Dockerfile.ubuntu [hunk 3]
```diff
@@ -37,18 +37,18 @@ RUN set -x && \
...
-# the sha256 sum can be found at https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl.sha256
+# the sha256 sum can be found at https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl.sha256
 # maybe a good idea to automate in the future?
-ENV KUBECTL_VERSION="v1.21.4"
-ENV KUBECTL_SHA256="9410572396fb31e49d088f9816beaebad7420c7686697578691be1651d3bf85a"
+ENV KUBECTL_VERSION="v1.25.2"
+ENV KUBECTL_SHA256="8639f2b9c33d38910d706171ce3d25be9b19fc139d0e3d4627f38ce84f9040eb"
 RUN set -x && \
-    curl --retry 5 --retry-connrefused -LO "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
+    curl --retry 5 --retry-connrefused -LO "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
```
Comment (yxxhero) [inline Dockerfile.ubuntu:45]:
```suggestion
    curl --retry 5 --retry-connrefused -LO "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
```

File: Dockerfile [hunk 4]
```diff
@@ -33,18 +33,18 @@ RUN set -x && \
...
-# the sha256 sum can be found at https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl.sha256
+# the sha256 sum can be found at https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl.sha256
 # maybe a good idea to automate in the future?
-ENV KUBECTL_VERSION="v1.21.4"
-ENV KUBECTL_SHA256="9410572396fb31e49d088f9816beaebad7420c7686697578691be1651d3bf85a"
+ENV KUBECTL_VERSION="v1.25.2"
+ENV KUBECTL_SHA256="8639f2b9c33d38910d706171ce3d25be9b19fc139d0e3d4627f38ce84f9040eb"
 RUN set -x && \
-    curl --retry 5 --retry-connrefused -LO "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
+    curl --retry 5 --retry-connrefused -LO "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
```
Comment (yxxhero) [inline Dockerfile:41]:
```suggestion
    curl --retry 5 --retry-connrefused -LO "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
```

File: Dockerfile [hunk 5]
```diff
@@ -33,18 +33,18 @@ RUN set -x && \
...
-    curl --retry 5 --retry-connrefused -LO "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
+    curl --retry 5 --retry-connrefused -LO "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
     sha256sum kubectl | grep ${KUBECTL_SHA256} && \
     chmod +x kubectl && \
     mv kubectl /usr/local/bin/kubectl
 
-ENV KUSTOMIZE_VERSION="v3.8.8"
-ENV KUSTOMIZE_SHA256="175938206f23956ec18dac3da0816ea5b5b485a8493a839da278faac82e3c303"
+ENV KUSTOMIZE_VERSION="v4.5.7"
+ENV KUSTOMIZE_SHA256="0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
```
Comment (yxxhero) [inline Dockerfile:47]:
```suggestion
ENV KUSTOMIZE_SHA256="701e3c4bfa14e4c520d481fdf7131f902531bfc002cb5062dcf31263a09c70c9"
```

File: Dockerfile.debian-stable-slim [hunk 6]
```diff
@@ -37,18 +37,18 @@ RUN set -x && \
...
-    wget "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
+    wget "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
     sha256sum kubectl | grep ${KUBECTL_SHA256} && \
     chmod +x kubectl && \
     mv kubectl /usr/local/bin/kubectl
 
-ENV KUSTOMIZE_VERSION="v3.8.8"
-ENV KUSTOMIZE_SHA256="175938206f23956ec18dac3da0816ea5b5b485a8493a839da278faac82e3c303"
+ENV KUSTOMIZE_VERSION="v4.5.7"
+ENV KUSTOMIZE_SHA256="0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
```
Comment (yxxhero) [inline Dockerfile.debian-stable-slim:51]:
```suggestion
ENV KUSTOMIZE_SHA256="701e3c4bfa14e4c520d481fdf7131f902531bfc002cb5062dcf31263a09c70c9"
```

File: Dockerfile.ubuntu [hunk 7]
```diff
@@ -37,18 +37,18 @@ RUN set -x && \
...
-    curl --retry 5 --retry-connrefused -LO "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
+    curl --retry 5 --retry-connrefused -LO "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" && \
     sha256sum kubectl | grep ${KUBECTL_SHA256} && \
     chmod +x kubectl && \
     mv kubectl /usr/local/bin/kubectl
 
-ENV KUSTOMIZE_VERSION="v3.8.8"
-ENV KUSTOMIZE_SHA256="175938206f23956ec18dac3da0816ea5b5b485a8493a839da278faac82e3c303"
+ENV KUSTOMIZE_VERSION="v4.5.7"
+ENV KUSTOMIZE_SHA256="0019dfc4b32d63c1392aa264aed2253c1e0c2fb09216f8e2cc269bbfb8bb49b5"
```
Comment (yxxhero) [inline Dockerfile.ubuntu:51]:
```suggestion
ENV KUSTOMIZE_SHA256="701e3c4bfa14e4c520d481fdf7131f902531bfc002cb5062dcf31263a09c70c9"
```

**Review by yxxhero — APPROVED** [review]

**Pull request merged**